### DISCUSSION TITLE
support for inferring webpackChunkName from filename

### DISCUSSION THREAD

**Discussion by dwelle** [webpack/webpack#12635] [Q&A]
Is there a plugin or configuration that allows for automatic chunk naming when dynamically importing, so that I don't have to pepper the code with `webpackChunkName` everywhere?

```js
const SomeComponent = React.lazy(async () => ({
  default: (
    await import(
      /* webpackChunkName: "SomeComponent" */ "./SomeComponent"
    )
  ).SomeComponent,
}));
```

**Comment by alexander-akait** [comment]
Use https://webpack.js.org/configuration/output/#outputchunkfilename as `Function`

**Reply by dwelle** [reply]
Thanks. Is there a reason why Webpack doesn't do this by default? Would be useful in cases where we can't change the config (CRA).

**Comment by sokra** [comment]
Webpack 5 creates named ids for chunks by default in development mode.

You can do that in production too if you want to: `optimization.chunkIds: "named"`
